### PR TITLE
added singleton SparkSessionProvider

### PR DESCRIPTION
**Pull request by Nitish1814**

### CODE REVIEW

File: spark/core/src/test/java/zingg/spark/core/TestImageType.java [hunk 1]
```diff
@@ -91,7 +92,7 @@ public void testUDFArray() {
 			df.printSchema();
 			// register ArrayDoubleSimilarityFunction as a UDF
 			TestUDFDoubleArr testUDFDoubleArr = new TestUDFDoubleArr();
-			SparkFnRegistrar.registerUDF2(spark, "testUDFDoubleArr", testUDFDoubleArr, DataTypes.DoubleType);
+			SparkFnRegistrar.registerUDF2(TestSparkBase.spark, "testUDFDoubleArr", testUDFDoubleArr, DataTypes.DoubleType);
```
Comment (sonalgoyal) [inline spark/core/src/test/java/zingg/spark/core/TestImageType.java:95]: why not inject spark instance variqable and not make changes throughout? The whole idea is the class code should not know or care about how spark session was created.

File: spark/core/src/test/java/zingg/spark/core/documenter/TestDocumenterBase.java [hunk 2]
```diff
@@ -14,21 +14,34 @@
...
+import zingg.common.client.ArgumentsUtil;
+import zingg.common.client.IArguments;
+import zingg.common.core.documenter.DocumenterBase;
+import zingg.common.core.documenter.TemplateFields;
+import zingg.spark.core.TestSparkBase;
 
 
-public class TestDocumenterBase extends ZinggSparkTester {
+@ExtendWith(TestSparkBase.class)
+public class TestDocumenterBase {
```
Comment (sonalgoyal) [inline spark/core/src/test/java/zingg/spark/core/documenter/TestDocumenterBase.java:35]: we should have a ganeric test and a spark specific test for these classes

**Review by sonalgoyal — CHANGES_REQUESTED** [review]

File: spark/core/src/test/java/zingg/spark/core/TestImageType.java (hunk 1 above)
Comment (Nitish1814) [reply spark/core/src/test/java/zingg/spark/core/TestImageType.java]: updated

File: spark/core/src/test/java/zingg/spark/core/documenter/TestDocumenterBase.java (hunk 2 above)
Comment (Nitish1814) [reply spark/core/src/test/java/zingg/spark/core/documenter/TestDocumenterBase.java:35]: issue created

**Pull request merged**